src/item_factory.cpp

```diff
@@ -884,6 +901,33 @@ void Item_factory::load_gunmod(JsonObject &jo)
     load_basic_info( jo, new_item_template );
 }
 
+void Item_factory::load( islot_magazine &slot, JsonObject &jo )
+{
+    slot.type = jo.get_string( "ammo_type" );
```

Comment:
It's will be great to see description of new entries used here in [doc/JSON_INFO.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md) but it can wait for another PR.